### ISSUE TITLE
Rule Change: `no-new` should allow `new Notification()`

### ISSUE THREAD

**Issue by homersimpsons** [eslint/eslint#15705]
### What rule do you want to change?

no-new

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
new Notification('A Notification.');
```

### What does the rule currently do for this code?

It triggers an error: "Do not use 'new' for side effects"

### What will the rule do after it's changed?

It won't trigger an error

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Relevant documentation: https://developer.mozilla.org/en-US/docs/Web/API/Notification/Notification

Another issue (in a different project): https://github.com/standard/standard/issues/1197

I tried to look if this issue had already been declared, but there are a lot of match (https://github.com/eslint/eslint/issues?q=is%3Aissue+no-new+Notification+is%3Aclosed) so I didn't go through all closed issues ...

I didn't tick the box for participation, the current code (https://github.com/eslint/eslint/blob/76a235a31718312c2ed202fdde030d329ca62486/lib/rules/no-new.js) does not seem simple to update without knowledge of eslint internals.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by homersimpsons** [comment]
@nzakas This seems to fit within the following quote:

> New ECMAScript features: We will also make sure stylistic rules are compatible with new ECMAScript features

In fact, this is the only way to create a `Notification`, so with this rule eslint will always complain for the ONLY valid way to create / display a notification.
